use nodejs v20

haydenhoang opened the pull request:
## Change Summary
<!--- Described your changes here -->
- Google map script tag is executed before the `initMap` is defined, which cause a blank page. Fixed by changing the script attribute from `async` to `defer`
![image](https://github.com/user-attachments/assets/7e829bd1-11dd-4b85-803f-f5af91a1e8f8)

## PR Checklist
<!--- Put an `x` inside the box : -->
- [x] I have read and signed the [Contributor License Agreement](https://forms.gle/PZyiY5N2GDQU8GsV9).

haydenhoang committed:
use node 20

jasonbosco reviewed (changes requested):
Seeing this deprecation warning:

```
yarn start
yarn run v1.22.22
$ parcel index.html --port 3000
Server running at http://localhost:3000
⠧ Building defer.js...
DEPRECATION WARNING: Passing percentage units to the global abs() function is deprecated.
In the future, this will emit a CSS abs() function to be resolved by the browser.
To preserve current behavior: math.abs(100%)
To emit a CSS abs() now: abs(#{100%})
More info: https://sass-lang.com/d/abs-percent

   ╷
54 │   $dividend: abs($dividend);
   │              ^^^^^^^^^^^^^^
   ╵
    node_modules/bootstrap/scss/vendor/_rfs.scss 54:14             divide()
    node_modules/bootstrap/scss/mixins/_grid.scss 66:15            row-cols()
    node_modules/bootstrap/scss/mixins/_grid-framework.scss 43:13  @content
    node_modules/bootstrap/scss/mixins/_breakpoints.scss 65:5      media-breakpoint-up()
    node_modules/bootstrap/scss/mixins/_grid-framework.scss 32:5   make-grid-columns()
    node_modules/bootstrap/scss/_grid.scss 72:3                    @import
    node_modules/bootstrap/scss/bootstrap.scss 16:9                @import
    src/bootstrap.scss 57:9                                        @import
    src/index.scss 1:9                                             root stylesheet
✨ Built in 2.12s
```

Could you also upgrade all dependencies to the latest version as part of the same PR?

jasonbosco committed:
Update caniuse-lite

haydenhoang committed:
fix: `initMap` is not a function, update deps

- Google map script tag is executed before the `initMap` is defined, which cause a blank page. Fixed by changing the script attribute from `async` to `defer`

jasonbosco reviewed (changes requested):
I see more warning in the browser console. Could you address those as well:

![Screenshot 2024-12-17 at 3 47 49 PM](https://github.com/user-attachments/assets/dc63a842-eb2a-436a-8b40-6ff7e05be45a)

haydenhoang committed:
migrate to bootstrap v5

haydenhoang committed:
migrate to popperjs v2

haydenhoang committed:
resolve typesensejs deprecation warnings

haydenhoang committed:
google map migrate to the Dynamic Library Import API, resolve instantsearchjs warnings

haydenhoang commented:
The google map deprecation warnings is from instantsearchjs so there is nothing we can do about it. There are also some minor UI changes when upgrading bootstrap from v4 to v5.
